### PR TITLE
Automated notification to update the efforts sheet before the end of billing cycle.

### PR DESCRIPTION
**Pull request by Himanshunegi0332**
Targets #2895 
### Description
Currently, we don't have any automated notification system feature which tells to update the efforts sheet of a project to avoid last minutes updates at the end of the billing cycle.
So, created an automated notification system telling all active project members  to update the efforts sheet to avoid last minutes updates.

### Checklist:
- [X] I have performed a self-review of my own code.

### CODE REVIEW

File: Modules/Project/Notifications/GoogleChat/NotificationToUpdateEffortForProject.php [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+    public function toGoogleChat($notifiable)
+    {
+        $projects = PROJECT::all();
+        foreach ($projects as $project) {
+            $diff_project = date_diff(($project->end_date), (CARBON::NOW()));
+
+            if ($diff_project->days == 1) {
+                return GoogleChatMessage::create()
+                    ->mentionAll('', " it's time to update your efforts, the end date is tomorrow!\n")
```
Comment (GauravGusain98) [inline Modules/Project/Notifications/GoogleChat/NotificationToUpdateEffortForProject.php:46]: This message is not matching the text that we discussed

File: Modules/Project/Notifications/GoogleChat/NotificationToUpdateEffortForProject.php [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        return [
+            GoogleChatChannel::class
+        ];
+    }
+
+    public function toGoogleChat($notifiable)
+    {
+        $projects = PROJECT::all();
+        foreach ($projects as $project) {
+            $diff_project = date_diff(($project->end_date), (CARBON::NOW()));
```
Comment (GauravGusain98) [inline Modules/Project/Notifications/GoogleChat/NotificationToUpdateEffortForProject.php:42]: Variable naming convention is not correct

Comment (GauravGusain98) [inline Modules/Project/Notifications/GoogleChat/NotificationToUpdateEffortForProject.php:42]:
```suggestion
            $diff_project = date_diff($project->end_date, today());
```

**Review by GauravGusain98 — CHANGES_REQUESTED** [review]
please look at comments

File: Modules/Project/Notifications/GoogleChat/NotificationToUpdateEffortForProject.php [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        return [
+            GoogleChatChannel::class
+        ];
+    }
+
+    public function toGoogleChat($notifiable)
+    {
+        $projects = Project::all();
+        foreach ($projects as $project) {
+            $diff_in_dates = date_diff($project->end_date, today());
```
Comment (GauravGusain98) [inline Modules/Project/Notifications/GoogleChat/NotificationToUpdateEffortForProject.php:36]: correct variable naming convention

**Review by GauravGusain98 — CHANGES_REQUESTED** [review]

File: Modules/Project/Notifications/GoogleChat/NotificationToUpdateEffortForProject.php (hunk 1 above)
Comment (Himanshunegi0332) [reply Modules/Project/Notifications/GoogleChat/NotificationToUpdateEffortForProject.php]: ok, Sorry for this.

File: Modules/Project/Notifications/GoogleChat/NotificationToUpdateEffortForProject.php (hunk 2 above)
Comment (Himanshunegi0332) [reply Modules/Project/Notifications/GoogleChat/NotificationToUpdateEffortForProject.php]: done with the changes

File: Modules/Project/Notifications/GoogleChat/NotificationToUpdateEffortForProject.php [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+                            ->section(
+                                Section::create(
+                                    KeyValue::create('Project', $this->project->name)
+                                    ->setContentMultiline(true)
+                                    ->icon(Icon::CLOCK)
+                                    ->button(
+                                        TextButton::create($this->project->effort_sheet_url, 'Open effortsheet')
+                                    )
+                                ),
+                            ),
```
[the remark this answers is not in the document]
Comment (Himanshunegi0332) [reply Modules/Project/Notifications/GoogleChat/NotificationToUpdateEffortForProject.php]: ok

File: Modules/Project/Notifications/GoogleChat/NotificationToUpdateEffortForProject.php (hunk 3 above)
Comment (Himanshunegi0332) [reply Modules/Project/Notifications/GoogleChat/NotificationToUpdateEffortForProject.php]: ok

File: Modules/Project/Console/GoogleChat/NotificationToProjectTeamMembersToUpdateEffortOnGoogleChat.php [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+use Modules\Project\Notifications\GoogleChat\NotificationToUpdateEffortForProject;
+
+class NotificationToProjectTeamMembersToUpdateEffortOnGoogleChat extends Command
+{
+    /**
+     * The console command name.
+     *
+     * @var string
+     */
+    protected $name = 'project:notification-to-update-effort-for-project';
```
Comment (GauravGusain98) [inline Modules/Project/Console/GoogleChat/NotificationToProjectTeamMembersToUpdateEffortOnGoogleChat.php:17]: Lets name it `project:reminder-for-effortsheet-lock`

Comment (GauravGusain98) [inline Modules/Project/Console/GoogleChat/NotificationToProjectTeamMembersToUpdateEffortOnGoogleChat.php:17]:
```suggestion
    protected $name = 'project:reminder-for-effortsheet-lock';
```

**Pull request merged**